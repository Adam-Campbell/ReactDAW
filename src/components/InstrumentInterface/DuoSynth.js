import React, { Component } from 'react';
import PropTypes from 'prop-types';
import HeaderModule from './HeaderModule';
import EnvelopeModule from './EnvelopeModule';
import OscillatorModule from './OscillatorModule';
import FilterModule from './FilterModule';
import FilterEnvelopeModule from './FilterEnvelopeModule';
import EnhancedRangeInput from '../EnhancedRangeInput';
import Dial from '../Dial';
import SmallDial from '../SmallDial';

class DuoSynth extends Component {
    constructor(props) {
        super(props);
        this.state = {
            isShowingVoiceA: true
        };
    }

    showVoiceA = () => {
        this.setState({
            isShowingVoiceA: true
        });
    }

    showVoiceB = () => {
        this.setState({
            isShowingVoiceA: false
        });
    };

    renderVoiceAControls() {
        return (
            <React.Fragment>
                <div className="instrument-interface__seperator">
                    <h2 className="instrument-interface__module-heading">Voice A</h2>
                </div>
                <EnvelopeModule 
                    envelopeData={this.props.instrumentData.voice0.envelope}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice0']}
                    dblCol
                />
                <OscillatorModule 
                    oscillatorData={this.props.instrumentData.voice0.oscillator}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice0']}
                    dblCol
                />
                <FilterModule 
                    filterData={this.props.instrumentData.voice0.filter}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice0']}
                />
                <FilterEnvelopeModule 
                    filterEnvelopeData={this.props.instrumentData.voice0.filterEnvelope}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice0']}
                />
            </React.Fragment>
        );
    }

    renderVoiceBControls() {
        return (
            <React.Fragment>
                <div className="instrument-interface__seperator">
                    <h2 className="instrument-interface__module-heading">Voice B</h2>
                </div>
                <EnvelopeModule 
                    envelopeData={this.props.instrumentData.voice1.envelope}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice1']}
                    dblCol
                />
                <OscillatorModule 
                    oscillatorData={this.props.instrumentData.voice1.oscillator}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice1']}
                    dblCol
                />
                <FilterModule 
                    filterData={this.props.instrumentData.voice1.filter}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice1']}
                />
                <FilterEnvelopeModule 
                    filterEnvelopeData={this.props.instrumentData.voice1.filterEnvelope}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice1']}
                />
            </React.Fragment>
        );
    }

    render() {

        return (
            <div className="instrument-interface__container">
                <HeaderModule instrumentTitle="Duo Synth" />
                <div className={
                    `instrument-interface__module-container 
                    instrument-interface__module-container--horizontal
                    instrument-interface__module-container--dbl-col`
                }>
                    <Dial
                        dataMin={0.25}
                        dataMax={8}
                        stepSize={0.05}
                        snapToStep={true}
                        value={this.props.instrumentData.harmonicity}
                        dialStartOffset={225}
                        dialRange={270}
                        updateValueCallback={(newVal) => this.props.handleChange(
                            this.props.instrumentId,
                            ['harmonicity'],
                            newVal
                        )}
                    >
                        {(props) => <SmallDial {...props} label="Harmonicity" />}
                    </Dial>
                    <Dial
                        dataMin={0}
                        dataMax={5}
                        stepSize={0.05}
                        snapToStep={true}
                        value={this.props.instrumentData.vibratoAmount}
                        dialStartOffset={225}
                        dialRange={270}
                        updateValueCallback={(newVal) => this.props.handleChange(
                            this.props.instrumentId,
                            ['vibratoAmount'],
                            newVal
                        )}
                    >
                        {(props) => <SmallDial {...props} label="Vibrato Amount" />}
                    </Dial>
                    <Dial
                        dataMin={5}
                        dataMax={1000}
                        stepSize={5}
                        snapToStep={true}
                        value={this.props.instrumentData.vibratoRate}
                        dialStartOffset={225}
                        dialRange={270}
                        updateValueCallback={(newVal) => this.props.handleChange(
                            this.props.instrumentId,
                            ['vibratoRate'],
                            newVal
                        )}
                    >
                        {(props) => <SmallDial {...props} label="Vibrato Rate" />}
                    </Dial>
                    <Dial
                        dataMin={-80}
                        dataMax={20}
                        stepSize={0.25}
                        snapToStep={true}
                        value={this.props.instrumentData.volume}
                        dialStartOffset={225}
                        dialRange={270}
                        updateValueCallback={(newVal) => this.props.handleChange(
                            this.props.instrumentId,
                            ['volume'],
                            newVal
                        )}
                    >
                        {(props) => <SmallDial {...props} label="Volume" />}
                    </Dial>
                </div>
                <div>
                    <button 
                        className="button off-white"
                        onClick={this.showVoiceA}
                    >Show Voice A</button>
                    <button 
                        className="button off-white"
                        onClick={this.showVoiceB}
                    >Show Voice B</button>
                </div>
                {this.state.isShowingVoiceA ? this.renderVoiceAControls() : this.renderVoiceBControls()}
            </div>
        )
    }
}

DuoSynth.propTypes = {
    instrumentData: PropTypes.object.isRequired,
    instrumentId: PropTypes.string.isRequired,
    handleChange: PropTypes.func.isRequired
};

export default DuoSynth;