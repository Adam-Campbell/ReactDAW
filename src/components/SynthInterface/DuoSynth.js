import React, { Component } from 'react';
import HeaderModule from './HeaderModule';
import EnvelopeModule from './EnvelopeModule';
import OscillatorModule from './OscillatorModule';
import FilterModule from './FilterModule';
import FilterEnvelopeModule from './FilterEnvelopeModule';
import RangeInput from './RangeInput';


class DuoSynth extends Component {
    constructor(props) {
        super(props);
        this.state = {
            isShowingVoiceA: true
        };
    }

    showVoiceA = () => {
        this.setState({
            isShowingVoiceA: true
        });
    }

    showVoiceB = () => {
        this.setState({
            isShowingVoiceA: false
        });
    };

    renderVoiceAControls() {
        return (
            <React.Fragment>
                <div className="synth-interface__seperator">
                    <h2 className="synth-interface__module-heading">Voice A</h2>
                </div>
                <EnvelopeModule 
                    envelopeData={this.props.instrumentData.voice0.envelope}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice0']}
                />
                <OscillatorModule 
                    oscillatorData={this.props.instrumentData.voice0.oscillator}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice0']}
                />
                <FilterModule 
                    filterData={this.props.instrumentData.voice0.filter}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice0']}
                />
                <FilterEnvelopeModule 
                    filterEnvelopeData={this.props.instrumentData.voice0.filterEnvelope}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice0']}
                />
            </React.Fragment>
        );
    }

    renderVoiceBControls() {
        return (
            <React.Fragment>
                <div className="synth-interface__seperator">
                    <h2 className="synth-interface__module-heading">Voice B</h2>
                </div>
                <EnvelopeModule 
                    envelopeData={this.props.instrumentData.voice1.envelope}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice1']}
                />
                <OscillatorModule 
                    oscillatorData={this.props.instrumentData.voice1.oscillator}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice1']}
                />
                <FilterModule 
                    filterData={this.props.instrumentData.voice1.filter}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice1']}
                />
                <FilterEnvelopeModule 
                    filterEnvelopeData={this.props.instrumentData.voice1.filterEnvelope}
                    handleChange={this.props.handleChange}
                    instrumentId={this.props.instrumentId}
                    additionalNesting={['voice1']}
                />
            </React.Fragment>
        );
    }

    render() {

        return (
            <div className="synth-interface__container">
                <HeaderModule synthTitle="Duo Synth" handleClose={this.props.handleClose} />
                <div className="synth-interface__module-container">
                    <RangeInput 
                        inputId={'harmonicity'}
                        label={'Harmonicity'}
                        min={0.25}
                        max={8}
                        step={0.25}
                        value={this.props.instrumentData.harmonicity}
                        handleChange={this.props.handleChange}
                        instrumentId={this.props.instrumentId}
                        propertyPathArray={['harmonicity']}
                    />
                    <RangeInput 
                        inputId={'vibrato-amount'}
                        label={'Vibrato Amount'}
                        min={0}
                        max={5}
                        step={0.2}
                        value={this.props.instrumentData.vibratoAmount}
                        handleChange={this.props.handleChange}
                        instrumentId={this.props.instrumentId}
                        propertyPathArray={['vibratoAmount']}
                    />
                    <RangeInput 
                        inputId={'vibrato-rate'}
                        label={'Vibrato Rate'}
                        min={5}
                        max={1000}
                        step={5}
                        value={this.props.instrumentData.vibratoRate}
                        handleChange={this.props.handleChange}
                        instrumentId={this.props.instrumentId}
                        propertyPathArray={['vibratoRate']}
                    />
                    <RangeInput 
                        inputId={'volume'}
                        label={'Volume'}
                        min={-80}
                        max={20}
                        step={1}
                        value={this.props.instrumentData.volume}
                        handleChange={this.props.handleChange}
                        instrumentId={this.props.instrumentId}
                        propertyPathArray={['volume']}
                    />
                </div>
                <div>
                    <button 
                        className="button off-white"
                        onClick={this.showVoiceA}
                    >Show Voice A</button>
                    <button 
                        className="button off-white"
                        onClick={this.showVoiceB}
                    >Show Voice B</button>
                </div>
                {this.state.isShowingVoiceA ? this.renderVoiceAControls() : this.renderVoiceBControls()}
            </div>
        )
    }
}



// const DuoSynth = props => (
//     <div className="synth-interface__container">
//         <HeaderModule synthTitle="Duo Synth" handleClose={props.handleClose} />
//         <div className="synth-interface__module-container">
//             <RangeInput 
//                 inputId={'harmonicity'}
//                 label={'Harmonicity'}
//                 min={0.25}
//                 max={8}
//                 step={0.25}
//                 value={props.instrumentData.harmonicity}
//                 handleChange={props.handleChange}
//                 instrumentId={props.instrumentId}
//                 propertyPathArray={['harmonicity']}
//             />
//             <RangeInput 
//                 inputId={'vibrato-amount'}
//                 label={'Vibrato Amount'}
//                 min={0}
//                 max={5}
//                 step={0.2}
//                 value={props.instrumentData.vibratoAmount}
//                 handleChange={props.handleChange}
//                 instrumentId={props.instrumentId}
//                 propertyPathArray={['vibratoAmount']}
//             />
//             <RangeInput 
//                 inputId={'vibrato-rate'}
//                 label={'Vibrato Rate'}
//                 min={5}
//                 max={1000}
//                 step={5}
//                 value={props.instrumentData.vibratoRate}
//                 handleChange={props.handleChange}
//                 instrumentId={props.instrumentId}
//                 propertyPathArray={['vibratoRate']}
//             />
//             <RangeInput 
//                 inputId={'volume'}
//                 label={'Volume'}
//                 min={-80}
//                 max={20}
//                 step={1}
//                 value={props.instrumentData.volume}
//                 handleChange={props.handleChange}
//                 instrumentId={props.instrumentId}
//                 propertyPathArray={['volume']}
//             />
//         </div>
//         <div className="synth-interface__seperator">
//             <h2 className="synth-interface__module-heading">Voice A</h2>
//         </div>
//         <EnvelopeModule 
//             envelopeData={props.instrumentData.voice0.envelope}
//             handleChange={props.handleChange}
//             instrumentId={props.instrumentId}
//             additionalNesting={['voice0']}
//         />
//         <OscillatorModule 
//             oscillatorData={props.instrumentData.voice0.oscillator}
//             handleChange={props.handleChange}
//             instrumentId={props.instrumentId}
//             additionalNesting={['voice0']}
//         />
//         <FilterModule 
//             filterData={props.instrumentData.voice0.filter}
//             handleChange={props.handleChange}
//             instrumentId={props.instrumentId}
//             additionalNesting={['voice0']}
//         />
//         <FilterEnvelopeModule 
//             filterEnvelopeData={props.instrumentData.voice0.filterEnvelope}
//             handleChange={props.handleChange}
//             instrumentId={props.instrumentId}
//             additionalNesting={['voice0']}
//         />
//         <div className="synth-interface__seperator">
//             <h2 className="synth-interface__module-heading">Voice B</h2>
//         </div>
//         <EnvelopeModule 
//             envelopeData={props.instrumentData.voice1.envelope}
//             handleChange={props.handleChange}
//             instrumentId={props.instrumentId}
//             additionalNesting={['voice1']}
//         />
//         <OscillatorModule 
//             oscillatorData={props.instrumentData.voice1.oscillator}
//             handleChange={props.handleChange}
//             instrumentId={props.instrumentId}
//             additionalNesting={['voice1']}
//         />
//         <FilterModule 
//             filterData={props.instrumentData.voice1.filter}
//             handleChange={props.handleChange}
//             instrumentId={props.instrumentId}
//             additionalNesting={['voice1']}
//         />
//         <FilterEnvelopeModule 
//             filterEnvelopeData={props.instrumentData.voice1.filterEnvelope}
//             handleChange={props.handleChange}
//             instrumentId={props.instrumentId}
//             additionalNesting={['voice1']}
//         />
//     </div>
// );

export default DuoSynth;