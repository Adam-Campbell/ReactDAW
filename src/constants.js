export const synthTypes = {
    default: 'synth',
    am: 'amSynth',
    fm: 'fmSynth',
    duo: 'duoSynth',
    mono: 'monoSynth'
};

export const UIColors = {
    deepPurple: '#201826',
    lightPurple: '#47426c',
    pink: '#d86597',
    offWhite: '#e0e0e0',
    brightBlue: '#00bff3',
    brightYellow: '#f4ab54',
    brightRed: '#e23340',
    brightGreen: '#25e452'
};

export const effectTypes = {
    autoFilter: 'autoFilter',
    bitCrusher: 'bitCrusher',
    chorus: 'chorus',
    compressor: 'compressor',
    distortion: 'distortion',
    eq3: 'eq3',
    feedbackDelay: 'feedbackDelay',
    filter: 'filter',
    freeverb: 'freeverb',
    gate: 'gate',
    jcReverb: 'jcReverb',
    lfo: 'lfo',
    limiter: 'limiter',
    multibandCompressor: 'multibandCompressor',
    phaser: 'phaser',
    pingPongDelay: 'pingPongDelay',
    pitchShift: 'pitchShift',
    tremolo: 'tremolo',
    vibrato: 'vibrato'
};

export const defaultSynthData = {
    envelope: {
        attack: 0.005,
        attackCurve: 'linear',
        decay: 0.1,
        release: 1,
        releaseCurve: 'exponential',
        sustain: 0.3
    },
    oscillator: {
        detune: 0,
        mute: false,
        phase: 0,
        type: 'triangle',
        volume: 0
    },
    portamento: 0,
    volume: 0
}

export const synthData = {
    synth: {
        envelope: {
            attack: 0.005,
            attackCurve: 'linear',
            decay: 0.1,
            release: 1,
            releaseCurve: 'exponential',
            sustain: 0.3
        },
        oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'triangle',
            volume: 0
        },
        portamento: 0,
        volume: 0
    },
    amSynth: {
        detune: 0,
        envelope: {
            attack: 0.01,
            attackCurve: 'linear',
            decay: 0.01,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
        },
        harmonicity: 3,
        modulation: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'square',
            volume: 0
        },
        modulationEnvelope: {
            attack: 0.5,
            attackCurve: 'linear',
            decay: 0,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
        },
        oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'triangle',
            volume: 0
        },
        portamento: 0,
        volume: 0
    },
    fmSynth: {
        detune: 0,
        envelope: {
            attack: 0.01,
            attackCurve: 'linear',
            decay: 0.01,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
        },
        harmonicity: 3,
        modulation: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'square',
            volume: 0
        },
        modulationEnvelope: {
            attack: 0.5,
            attackCurve: 'linear',
            decay: 0,
            release: 0.5,
            releaseCurve: 'exponential',
            sustain: 1
        },
        modulationIndex: 10,
        oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'triangle',
            volume: 0
        },
        portamento: 0,
        volume: 0
    },
    duoSynth: {
        harmonicity: 1.5,
        portamento: 0,
        vibratoAmount: 0.5,
        vibratoRate: 5,
        voice0: {
            detune: 0,
            envelope: {
                attack: 0.01,
                attackCurve: 'linear',
                decay: 0,
                release: 0.5,
                releaseCurve: 'exponential',
                sustain: 1
            },
            filter: {
                Q: 6,
                frequency: 350,
                gain: 0,
                rolloff: -24,
                type: 'lowpass'
            },
            filterEnvelope: {
                attack: 0.01,
                attackCurve: 'linear',
                baseFrequency: 200,
                decay: 0,
                exponent: 2,
                octaves: 7,
                release: 0.5,
                releaseCurve: 'exponential',
                sustain: 1
            },
            oscillator: {
                detune: 0,
                mute: false,
                phase: 0,
                type: 'sine',
                volume: 0
            },
            portamento: 0,
            volume: -9.999999999999998
        },
        voice1: {
            detune: 0,
            envelope: {
                attack: 0.01,
                attackCurve: 'linear',
                decay: 0,
                release: 0.5,
                releaseCurve: 'exponential',
                sustain: 1
            },
            filter: {
                Q: 6,
                frequency: 350,
                gain: 0,
                rolloff: -24,
                type: 'lowpass'
            },
            filterEnvelope: {
                attack: 0.01,
                attackCurve: 'linear',
                baseFrequency: 200,
                decay: 0,
                exponent: 2,
                octaves: 7,
                release: 0.5,
                releaseCurve: 'exponential',
                sustain: 1
            },
            oscillator: {
                detune: 0,
                mute: false,
                phase: 0,
                type: 'sine',
                volume: 0
            },
            portamento: 0,
            volume: -9.999999999999998
        },
        volume: 0
    },
    monoSynth: {
        detune: 0,
        envelope: {
            attack: 0.005,
            attackCurve: 'linear',
            decay: 0.1,
            release: 1,
            releaseCurve: 'exponential',
            sustain: 0.9
        },
        filter: {
            Q: 6,
            frequency: 350,
            gain: 0,
            rolloff: -24,
            type: 'lowpass'
        },
        filterEnvelope: {
            attack: 0.06,
            attackCurve: 'linear',
            baseFrequency: 200,
            decay: 0.2,
            exponent: 2,
            octaves: 7,
            release: 2,
            releaseCurve: 'exponential',
            sustain: 0.5
        },
        oscillator: {
            detune: 0,
            mute: false,
            phase: 0,
            type: 'square',
            volume: 0
        },
        portamento: 0,
        volume: 0
    }
};


export const effectData = {
    autoFilter: {
        baseFrequency: 200,
        depth: 1,
        filter: {
            Q: 1,
            gain: 0,
            rolloff: -12,
            type: 'lowpass'
        },
        frequency: 1,
        octaves: 2.6,
        type: 'sine',
        wet: 1
    },
    bitCrusher: {
        bits: 4,
        wet: 1
    },
    chorus: {
        delayTime: 3.5,
        depth: 0.7,
        frequency: 1.5,
        spread: 180,
        type: 'sine',
        wet: 1
    },
    compressor: {
        attack: 0.003,
        knee: 30,
        ratio: 12,
        release: 0.25,
        threshold: -24
    },
    distortion: {
        distortion: 0.4,
        oversample: 'none',
        wet: 1
    },
    eq3: {
        high: 0,
        highFrequency: 2500,
        low: 0,
        lowFrequency: 400,
        mid: 0
    },
    feedbackDelay: {
        delayTime: 0.25,
        feedback: 0.125,
        wet: 1
    },
    filter: {
        Q: 1,
        frequency: 350,
        gain: 0,
        rolloff: -12,
        type: 'lowpass'
    },
    freeverb: {
        dampening: 3000,
        roomSize: 0.7,
        wet: 1
    },
    gate: {
        attack: 0.1, 
        release: 0.1,
        threshold: -39.99999999999999
    },
    jcReverb: {
        roomSize: 0.5,
        wet: 1,
    },
    lfo: {
        amplitude: 1,
        frequency: 2,
        max: 1,
        min: 0,
        phase: 0,
        type: 'sine',
        units: 'number'
    },
    limiter: {
        threshold: -12
    },
    multibandCompressor: {
        high: {
            attack: 0.003,
            knee: 30,
            ratio: 12,
            release: 0.25,
            threshold: -24
        },
        highFrequency: 2500,
        low: {
            attack: 0.003,
            knee: 30,
            ratio: 12,
            release: 0.25,
            threshold: -24
        },
        lowFrequency: 400,
        mid: {
            attack: 0.003,
            knee: 30,
            ratio: 12,
            release: 0.25,
            threshold: -24
        }
    },
    phaser: {
        Q: 10,
        baseFrequency: 350,
        frequency: 0.5,
        octaves: 3,
        wet: 1
    },
    pingPongDelay: {
        delayTime: 0.25,
        wet: 1
    },
    pitchShift: {
        delayTime: 0,
        feedBack: 0,
        pitch: 0,
        wet: 1,
        windowSize: 0.1
    },
    tremolo: {
        depth: 0.5,
        frequency: 10,
        spread: 180,
        type: 'sine',
        wet: 1
    },
    vibrato: {
        depth: 0.1,
        frequency: 5,
        type: 'sine',
        wet: 1
    }
}