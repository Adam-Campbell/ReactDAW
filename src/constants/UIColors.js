// export const UIColors = {
//     deepPurple: '#030303',
//     lightPurple: '#131313',
//     pink: '#23af62',
//     offWhite: '#e0e0e0',
//     brightBlue: '#00bff3',
//     brightYellow: '#f4ab54',
//     brightRed: '#e23340',
//     brightGreen: '#25e452',
//     darkGrey: '#222222'
// };

export const UIColors = {
    main: '#23af62',
    mainAlt: '#1d9653',
    backgroundDark: '#030303',
    backgroundMedium: '#131313',
    backgroundLight: '#232323',
    offWhite: '#e0e0e0',
    offWhiteAlt: '#ccc9c9',
    brightBlue: '#00bff3',
    brightRed: '#e23340',
    brightYellow: '#f4ab54',
    pink: '#d86597'
}

// background gradient overlay: linear-gradient(transparent, 60%, #0b0b0b);

// medium old

// export const UIColors = {
//     deepPurple: '#14161c',
//     lightPurple: '#252f48',
//     pink: '#aa9c9c',
//     offWhite: '#e0e0e0',
//     brightBlue: '#00bff3',
//     brightYellow: '#f4ab54',
//     brightRed: '#e23340',
//     brightGreen: '#25e452',
//     darkGrey: '#222222'
// };


// oldest old

// export const UIColors = {
//     deepPurple: '#201826',
//     lightPurple: '#47426c',
//     pink: '#d86597',
//     offWhite: '#e0e0e0',
//     brightBlue: '#00bff3',
//     brightYellow: '#f4ab54',
//     brightRed: '#e23340',
//     brightGreen: '#25e452',
//     darkGrey: '#222222'
// };
