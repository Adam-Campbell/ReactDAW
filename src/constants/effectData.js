export const effectData = {
    autoFilter: {
        baseFrequency: 200,
        depth: 1,
        filter: {
            Q: 1,
            gain: 0,
            rolloff: -12,
            type: 'lowpass'
        },
        frequency: 1,
        octaves: 2.6,
        type: 'sine',
        wet: 1
    },
    bitCrusher: {
        bits: 4,
        wet: 1
    },
    chorus: {
        delayTime: 3.5,
        depth: 0.7,
        frequency: 1.5,
        spread: 180,
        type: 'sine',
        wet: 1
    },
    compressor: {
        attack: 0.003,
        knee: 30,
        ratio: 12,
        release: 0.25,
        threshold: -24
    },
    distortion: {
        distortion: 0.4,
        oversample: 'none',
        wet: 1
    },
    eq3: {
        high: 0,
        highFrequency: 2500,
        low: 0,
        lowFrequency: 400,
        mid: 0
    },
    feedbackDelay: {
        delayTime: '8n',
        feedback: 0.125,
        wet: 1
    },
    filter: {
        Q: 1,
        frequency: 350,
        gain: 0,
        rolloff: -12,
        type: 'lowpass'
    },
    freeverb: {
        dampening: 3000,
        roomSize: 0.7,
        wet: 1
    },
    gate: {
        attack: 0.1, 
        release: 0.1,
        threshold: -39.99999999999999
    },
    jcReverb: {
        roomSize: 0.5,
        wet: 1,
    },
    lfo: {
        amplitude: 1,
        frequency: 2,
        max: 1,
        min: 0,
        phase: 0,
        type: 'sine',
        units: 'number'
    },
    limiter: {
        threshold: -12
    },
    multibandCompressor: {
        high: {
            attack: 0.003,
            knee: 30,
            ratio: 12,
            release: 0.25,
            threshold: -24
        },
        highFrequency: 2500,
        low: {
            attack: 0.003,
            knee: 30,
            ratio: 12,
            release: 0.25,
            threshold: -24
        },
        lowFrequency: 400,
        mid: {
            attack: 0.003,
            knee: 30,
            ratio: 12,
            release: 0.25,
            threshold: -24
        }
    },
    phaser: {
        Q: 10,
        baseFrequency: 350,
        frequency: 0.5,
        octaves: 3,
        wet: 1
    },
    pingPongDelay: {
        delayTime: 0.25,
        wet: 1
    },
    pitchShift: {
        delayTime: 0,
        feedBack: 0,
        pitch: 0,
        wet: 1,
        windowSize: 0.1
    },
    tremolo: {
        depth: 0.5,
        frequency: 10,
        spread: 180,
        type: 'sine',
        wet: 1
    },
    vibrato: {
        depth: 0.1,
        frequency: 5,
        type: 'sine',
        wet: 1
    }
};
