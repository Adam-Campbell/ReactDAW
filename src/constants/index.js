export * from './UIColors';
export * from './programData';
export * from './instrumentData';
export * from './effectData';