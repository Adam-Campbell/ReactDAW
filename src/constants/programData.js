export const modalTypes = {
    open: 'Open',
    saveAs: 'Save As',
    help: 'help'
};

export const toolTypes = {
    cursor: 'cursor',
    pencil: 'pencil',
    selection: 'selection'
};

export const snapSettings = {
    _32t: '32t',
    _32n: '32n',
    _16t: '16t',
    _16n: '16n',
    _8t: '8t',
    _8n: '8n',
    _4t: '4t',
    _4n: '4n',
    _2t: '2t',
    _2n: '2n',
    _1m: '1m'
};

export const noteDurationSettings = {
    _32t: '32t',
    _32n: '32n',
    _16t: '16t',
    _16n: '16n',
    _8t: '8t',
    _8n: '8n',
    _4t: '4t',
    _4n: '4n',
    _2t: '2t',
    _2n: '2n',
    _1m: '1m'
};

export const instrumentTypes = {
    default: 'synth',
    am: 'amSynth',
    fm: 'fmSynth',
    duo: 'duoSynth',
    mono: 'monoSynth',
    drumKit: 'drumKit'
};

export const effectTypes = {
    autoFilter: 'autoFilter',
    bitCrusher: 'bitCrusher',
    chorus: 'chorus',
    compressor: 'compressor',
    distortion: 'distortion',
    eq3: 'eq3',
    feedbackDelay: 'feedbackDelay',
    filter: 'filter',
    freeverb: 'freeverb',
    gate: 'gate',
    jcReverb: 'jcReverb',
    lfo: 'lfo',
    limiter: 'limiter',
    multibandCompressor: 'multibandCompressor',
    phaser: 'phaser',
    pingPongDelay: 'pingPongDelay',
    pitchShift: 'pitchShift',
    tremolo: 'tremolo',
    vibrato: 'vibrato'
};