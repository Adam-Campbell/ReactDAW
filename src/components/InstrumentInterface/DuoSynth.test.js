import React from 'react';
import { shallow } from 'enzyme';
import DuoSynth from './DuoSynth';
import { instrumentData } from '../../constants';

test('renders correctly', () => {
    const component = shallow(
        <DuoSynth 
            instrumentData={instrumentData.duoSynth}
            instrumentId="5542150612118159"
            handleChange={jest.fn()}
        />
    );
    expect(component).toMatchSnapshot();
});