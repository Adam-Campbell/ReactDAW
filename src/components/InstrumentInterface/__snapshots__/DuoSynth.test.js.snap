// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="synth-interface__container"
>
  <HeaderModule
    synthTitle="Duo Synth"
  />
  <div
    className="synth-interface__module-container"
  >
    <EnhancedRangeInput
      handleChange={[MockFunction]}
      identifier="5542150612118159"
      inputId="harmonicity"
      label="Harmonicity"
      max={8}
      min={0.25}
      propertyPathArray={
        Array [
          "harmonicity",
        ]
      }
      step={0.25}
      value={1.5}
    />
    <EnhancedRangeInput
      handleChange={[MockFunction]}
      identifier="5542150612118159"
      inputId="vibrato-amount"
      label="Vibrato Amount"
      max={5}
      min={0}
      propertyPathArray={
        Array [
          "vibratoAmount",
        ]
      }
      step={0.2}
      value={0.5}
    />
    <EnhancedRangeInput
      handleChange={[MockFunction]}
      identifier="5542150612118159"
      inputId="vibrato-rate"
      label="Vibrato Rate"
      max={1000}
      min={5}
      propertyPathArray={
        Array [
          "vibratoRate",
        ]
      }
      step={5}
      value={5}
    />
    <EnhancedRangeInput
      handleChange={[MockFunction]}
      identifier="5542150612118159"
      inputId="volume"
      label="Volume"
      max={20}
      min={-80}
      propertyPathArray={
        Array [
          "volume",
        ]
      }
      step={1}
      value={0}
    />
  </div>
  <div>
    <button
      className="button off-white"
      onClick={[Function]}
    >
      Show Voice A
    </button>
    <button
      className="button off-white"
      onClick={[Function]}
    >
      Show Voice B
    </button>
  </div>
  <div
    className="synth-interface__seperator"
  >
    <h2
      className="synth-interface__module-heading"
    >
      Voice A
    </h2>
  </div>
  <EnvelopeModule
    additionalNesting={
      Array [
        "voice0",
      ]
    }
    envelopeData={
      Object {
        "attack": 0.01,
        "attackCurve": "linear",
        "decay": 0,
        "release": 0.5,
        "releaseCurve": "exponential",
        "sustain": 1,
      }
    }
    handleChange={[MockFunction]}
    instrumentId="5542150612118159"
  />
  <OscillatorModule
    additionalNesting={
      Array [
        "voice0",
      ]
    }
    handleChange={[MockFunction]}
    instrumentId="5542150612118159"
    oscillatorData={
      Object {
        "detune": 0,
        "mute": false,
        "phase": 0,
        "type": "sine",
        "volume": 0,
      }
    }
  />
  <FilterModule
    additionalNesting={
      Array [
        "voice0",
      ]
    }
    filterData={
      Object {
        "Q": 6,
        "frequency": 350,
        "gain": 0,
        "rolloff": -24,
        "type": "lowpass",
      }
    }
    handleChange={[MockFunction]}
    instrumentId="5542150612118159"
  />
  <FilterEnvelopeModule
    additionalNesting={
      Array [
        "voice0",
      ]
    }
    filterEnvelopeData={
      Object {
        "attack": 0.01,
        "attackCurve": "linear",
        "baseFrequency": 200,
        "decay": 0,
        "exponent": 2,
        "octaves": 7,
        "release": 0.5,
        "releaseCurve": "exponential",
        "sustain": 1,
      }
    }
    handleChange={[MockFunction]}
    instrumentId="5542150612118159"
  />
</div>
`;
